python/lsst/analysis/tools/actions/plot/histPlot.py

```diff
@@ -87,19 +87,33 @@ class HistStatsPanel(Config):
     stat1 = ListField[str](
         doc="A list specifying the vector keys of the first scalar statistic to be shown in this panel."
         "there should be one entry for each hist in the panel",
-        default=[],
+        default=None,
+        optional=True,
     )
     stat2 = ListField[str](
         doc="A list specifying the vector keys of the second scalar statistic to be shown in this panel."
         "there should be one entry for each hist in the panel",
-        default=[],
+        default=None,
+        optional=True,
     )
     stat3 = ListField[str](
         doc="A list specifying the vector keys of the third scalar statistic to be shown in this panel."
         "there should be one entry for each hist in the panel",
-        default=[],
+        default=None,
+        optional=True,
     )
 
+    def validate(self):
+        super().validate()
+        if not all([self.stat1, self.stat2, self.stat3]) and any([self.stat1, self.stat2, self.stat3]):
+            msg = "If one stat is configured, all 3 stats must be configured"
+            if not bool(self.stat1):
+                raise FieldValidationError(self.__class__.stat1, self, msg)
+            if not bool(self.stat2):
+                raise FieldValidationError(self.__class__.stat2, self, msg)
+            if not bool(self.stat3):
+                raise FieldValidationError(self.__class__.stat3, self, msg)
+
```

Comment:
Correct, I had hoped the error message when a validation fails `If one stat is configured, all 3 stats must be configured`, would be enough, but I am open to suggestion on how to improve this. Would you recommend checking them all and setting a flag if they fail, and then returning an error that specifies which ones fail. 

They are fixed at 3 because of how the statsPanel was implemented in hist plot:
https://github.com/lsst/analysis_tools/blob/89b75f296be3630290e446a48cc85852c5ab1226/python/lsst/analysis/tools/actions/plot/histPlot.py#L583-L591

I think format was chosen to get the right spacing and behavior for the 3 column default case, but breaks if more or fewer columns are provided, and does not easily generalize. 

I called them `stat1/2/3` because they are supposed to be very general, I wanted to make an easy way showing metrics (such as PA1 and PF1) on a hist plot. I also wanted to make sure the metric calculation used the same code as the stat panel in the histogram; this is why the stats are computed as a build action and passed to hist plot for the produce stage. I am open to calling them something else, but am not sure what would be better.

Reply:
I leave this all up to you (i.e. I don't really have better suggestions that wouldn't involve a level of verbosity along the lines of your explanations above that some may not like...)!  Personally, I'd prefer just to get the message that `If one stat is configured, all 3 stats must be configured` with no attempt to identify which ones are invalid as opposed to a partial (and thus potentially misleading) list (i.e. specify all or none).

Reply:
Unfortunately pex.config.FieldValidationError requires a specific field to be raised (and can only mention 1 field), so I implemented a ValueError in the validation portion with a generic message example here: 
`ValueError: plots.skySourceHistPlot.produce.panels['panel_sn'].statsPanel : If one stat is configured, all 3 stats must be configured`